src/components/SubNav/SubNav.tsx

```diff
@@ -0,0 +1,312 @@
+ * actions related to the current section of a website.
+ */
+export const SubNav: ChakraComponent<
+  React.ForwardRefExoticComponent<
+    React.PropsWithChildren<SubNavProps> & React.RefAttributes<HTMLDivElement>
+  >,
+  SubNavProps
+> = chakra(
+  forwardRef<HTMLDivElement, React.PropsWithChildren<SubNavProps>>(
+    (props, _ref?) => {
+      const {
+        className,
+        actionBackgroundColor,
+        highlightColor,
+        primaryActions,
+        secondaryActions,
+      } = props;
+
+      const [showRightFade, setShowRightFade] = useState(false);
+      const [lastScrollLeft, setLastScrollLeft] = useState(0);
+      const scrollableRef = useRef(null);
+
+      if (actionBackgroundColor !== undefined && highlightColor === undefined) {
+        console.warn(
+          "NYPL Reservoir SubNav: The `actionBackgroundColor` prop has been passed, but the `highlightColor` prop has not been passed. Because of this, the `actionBackgroundColor` prop will be ignored."
+        );
+      }
+
+      const validateActions = (actions: React.ReactNode, propName: string) => {
+        if (React.isValidElement(actions)) {
+          React.Children.forEach(
+            actions.props.children,
+            (child: React.ReactElement) => {
+              if (child.type !== SubNavButton && child.type !== SubNavLink) {
+                console.warn(
+                  `NYPL Reservoir SubNav: An element that is not a SubNavButton or SubNavLink component has been passed in the \`${propName}\` prop. That element will be ignored.`
+                );
+                return null;
+              }
+            }
+          );
+        }
+      };
+
+      // Validate primaryActions and secondaryActions
+      validateActions(primaryActions, "primaryActions");
+      validateActions(secondaryActions, "secondaryActions");
+
+      const backgroundColor =
+        highlightColor !== undefined ? actionBackgroundColor : undefined;
+
+      const styles = useMultiStyleConfig("SubNav", {
+        backgroundColor: backgroundColor,
+        highlightColor: highlightColor,
+      });
+
+      const handleScroll = useCallback(() => {
+        if (scrollableRef.current) {
+          const { scrollLeft, scrollWidth, clientWidth } =
+            scrollableRef.current;
+
+          // Determine if we are scrolling to the left or right
+          const isScrollingRight = scrollLeft > lastScrollLeft;
+          setLastScrollLeft(scrollLeft); // Update last scroll position
+
+          // Show right fade only if scrolling to the left
+          if (!isScrollingRight) {
+            // Show right fade only if not at the end
+            const atRightEdge = scrollLeft + clientWidth >= scrollWidth;
+            setShowRightFade(!atRightEdge);
+          } else {
+            setShowRightFade(false); // Hide when scrolling right
+          }
+        }
+      }, [lastScrollLeft, setLastScrollLeft, setShowRightFade]);
+
+      useEffect(() => {
```

Comment:
This is great. This should be a reusable hook instead of declaring it inside this component. Doing so helps separate the component UI with the functionality/styles. This fading style is also used in the `Table` so it might overlap there. For now, you can create a hook file and only apply it to this component.